identity-hub-client/src/test/java/org/eclipse/dataspaceconnector/identityhub/client/IdentityHubClientImplTest.java

```diff
@@ -138,17 +143,34 @@ void addVerifiableCredentialsServerError() throws Exception {
         };
 
         var client = createClient(interceptor);
-        var statusResult = client.addVerifiableCredential(HUB_URL, credential);
+        var statusResult = client.addVerifiableCredential(HUB_URL, jws);
 
         var expectedResult = StatusResult.failure(ResponseStatus.FATAL_ERROR, String.format("IdentityHub error response code: %s, response headers: , response body: %s", code, body));
         assertThat(statusResult).usingRecursiveComparison().isEqualTo(expectedResult);
     }
 
+    @Test
+    void addVerifiableCredentialsIoException() throws Exception {
+        var credential = VerifiableCredential.Builder.newInstance().id(VERIFIABLE_CREDENTIAL_ID).build();
+        var jws = buildSignedJwt(credential, FAKER.internet().url(), FAKER.internet().url());
+        var exceptionMessage = "Can't resolve address";
```

Comment:
use Faker here (to ensure the implementation doesn't "accidentally" pass the test by having the same error message coded for some other condition)